### PR TITLE
Ditto Archetype as IPublishedContent

### PR DESCRIPTION
**Pull request by leekelleher**
Reworked the Ditto/Archetype approach.

Experimenting with converting `ArchetypeModel` to an `IPublishedContent` object, then using Ditto to map it.

### CODE REVIEW

**Comment by mattbrailsford** [comment]
Samples! Nice work.

**Comment by leekelleher** [comment]
@mattbrailsford

> Samples!

Is that a new buzzword like "totes"? or were you meaning the unit-tests? LOLz haha :laughing:

File: src/Our.Umbraco.Ditto.Archetype/Models/ArchetypePublishedProperty.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            }
+        }
+
+        public object Value
+        {
+            get
+            {
+                if (this.HasValue)
+                {
+                    return _property.Value;
```
Comment (leekelleher) [inline src/Our.Umbraco.Ditto.Archetype/Models/ArchetypePublishedProperty.cs:52]: I'm struggling with what to return here... whether it should be the raw `_property.Value`, or `_property.GetValue<T>()`?

The problem being that at this point we don't know what the target type should be.

If we go with the raw value, then we skip Archetype's call to `TryConvertWithPropertyValueConverters`...
https://github.com/imulus/Archetype/blob/master/app/Umbraco/Umbraco.Archetype/Models/ArchetypePropertyModel.cs#L39

So far my unit-tests have been with simple `string` values.  I'll aim to try with more complex data models.

**Comment by leekelleher** [comment]
Merged in, I think it's the right approach for Ditto. Thanks @mattbrailsford for re-prompting the idea!

**Pull request merged**